### DISCUSSION TITLE
What Are We Building?

### DISCUSSION THREAD

**Discussion by sethdavis512** [remix-austin/remixaustin-com#1] [Ideas]
What will the initial approach be for this site? A bunch of `.mdx` files to get the ball rolling? How complicated do we want things to get? We gonna use a Remix stack? Are we trying to replicate Meetup.com? Mostly curious - thanks ya'll!

**Comment by brookslybrand** [comment]
Hey @sethdavis512! Thanks for snooping around and starting some conversations. I've been thinking about this myself, and I think @grantnorwood  has some opinions too.

At the very minimum, I'd love for this site to be a nice single page with good SEO to drive more organic traffic to the meetup. Seems like there are ideas to include blogs as well, which could be cool, but definitely warrants further discussion around how we would want to do that.

> Are we trying to replicate Meetup.com

Great question 😅. I'm open to hearing what others think, but at the moment I'm very much in a "let's crawl before we run" mindset

**Reply by sethdavis512** [reply]
Agreed - Crawl then run.

What's y'alls stance on contribution? Mind if I send up a branch? If so could I get permissions?

**Reply by brookslybrand** [reply]
> What's y'alls stance on contribution? Mind if I send up a branch? If so could I get permissions?

Oh yeah, for sure! I'll grant you permission now. We don't have many rules at the moment, eventually I'd like to get to a place where it's pretty obvious what needs to be done via issue. I'd say at this point just create an issue with a description of what you're wanting to do along with PR if it's no trouble, will just help a bit with setting precedent

**Comment by grantnorwood** [comment]
re: SEO ... I've created a [PR](https://github.com/remix-austin/remixaustin-com/pull/31) to add some basic SEO improvements, and would love some feedback on what else we can do to make our site more discoverable in SERPs.

**Comment by brookslybrand** [comment]
I want to discuss what we want to do for content creation.

@KarenDouglas brought up a few great ["post" ideas](https://github.com/remix-austin/remixaustin-com/discussions/27#discussioncomment-4285968), such as "why you should present" or "Why you should attend the meetup"

I think these are really great, and I imagine we'll want a _lot_ more posts and written content on this sight. Building each of these pages in HTML probably isn't the best way to go. Definitely a big fan of the idea of using [mdx](https://mdxjs.com/).

Remix does [support `.mdx` files as routes](https://remix.run/docs/en/v1/guides/mdx), though even in their documentation, they don't encourage this (at least long term)

> Rather than compiling your content at build-time like this document demonstrates, it's typically better UX and DX if you do this at runtime via something like [mdx-bundler](https://github.com/kentcdodds/mdx-bundler). It's also much more customizable and powerful. However, if you prefer to do this compilation at build-time, continue reading.

Curious everyone's thoughts here? I think we have a few options:
- Start with builtin mdx-routes, and move to someone else later on
- Use GitHub as our CMS (this is what Kent does for his website -- allows the posts to still be open source but doesn't have to run the mdx-to-html build step every time)
- Leverage a CMS (gotta get [Paul](https://twitter.com/codingthirty) involved, I'm sure he'd love to recommend [strapi](https://strapi.io/))

c.c. @grantnorwood @sethdavis512 @tcschiller @KarenDouglas

**Reply by sethdavis512** [reply]
I've been doing some digging and found this [very interesting post on embedding Sanity into Remix](https://www.sanity.io/guides/remix-run-live-preview) which apparently is a dated tutorial BUT they included a link to a repo that includes a [more up-to-date approach](https://github.com/SimeonGriggs/remix-sanity-studio-v3) using [Sanity's preview kit](https://github.com/sanity-io/preview-kit).

<img width="587" alt="image" src="https://user-images.githubusercontent.com/3208705/205517342-3e4dbada-e4b6-498c-aa0d-a1a7bf411613.png">

I'm very intrigued and will continue to go down this trail...if it's not clear, my suggestion is that we go with a CMS like Sanity to get the ball rolling.

**Reply by grantnorwood** [reply]
So I found a coupon code which will hopefully [double our Sanity free plan limits](https://www.sanity.io/syntax), and created an account to play around.

It seems even more mature than it was last time I played with Sanity over a year ago, but it's also **$100/mo 💵** for a Team plan!  If we were to grow into needing a paid plan, we'd have to locate some budget 🙂

I did shoot an email to them and asked about sponsorships & other plans on the cheap for OSS projects, so perhaps I'll hear from them shortly on that.

@sethdavis512 Feel free to explore Sanity.  I'd be curious if their [GROG query language](https://www.sanity.io/docs/groq) is any good.  They also have a [GraphQL api](https://www.sanity.io/docs/graphql) for projects.  [Preview Kit](https://github.com/sanity-io/preview-kit) looks really neat, too!

I'm also looking at [Strapi](https://strapi.io/).  With our newfound CI/CD capabilities, it wouldn't be too difficult to manage a CMS admin dashboard app alongside our website.  And free sounds nice 😆  Although we'll potentially rack up some additional hosting costs, but we don't have to stay with our host if it gets expensive _(especially because it's on my 💳 !)_

As a technical requirement, I'd love to be able to still author content in Markdown whatever we choose.  And have previews _(even if we need to do some implementation/automation ourselves)_.

**Reply by brookslybrand** [reply]
Nice! Yeah Sanity seems pretty cool from what I've seen. For sure a little nervous about potential future costs. I'm all for starting with a free version if that gets the ball rolling, and then migrating away to something free, cheeper, or even building our own (as an dev who wants to over-engineer every problem, that sounds way more fun 🤓)

Anyway, at the least using Sanity to start could give us an idea for APIs/abstractions we'd want to build out for this site, so lets just think in terms of "how can we write this so that we can replace the CMS easily down the road".

Also, thanks @grantnorwood for reaching out to the Sanity folks to see if they have other plans/sponsorship for something like this project. Wouldn't mind even advertising them directly on the site if that helped (e.g. "Built with Remix + Sanity")

**Reply by KarenDouglas** [reply]
> @KarenDouglas brought up a few great https://github.com/remix-austin/remixaustin-com/discussions/27#discussioncomment-4285968, such as "why you should present" or "Why you should attend the meetup"
I think these are really great, and I imagine we'll want a lot more posts and written content on this sight. Building each of these pages in HTML probably isn't the best way to go. Definitely a big fan of the idea of using [mdx](https://mdxjs.com/).

I can get started on a draft for "why you should attend the meetup" in the meantime. 

> Start with builtin mdx-routes, and move to someone else later on

Seems like a good idea since we don't have a lot of content right now.

**Reply by sethdavis512** [reply]
> Even building our own (as an dev who wants to over-engineer every problem, that sounds way more fun 🤓)

@brookslybrand - I legit had the same thought. Build a "tiny CMS" in our instance of Remix 💯 but what if we took that a step further...make our own "CMS" and then contributed it back to the Remix community? Could be a very interesting idea...

**Reply by grantnorwood** [reply]
FYI, I heard back from Sanity on OSS licenses, and we can apply!

Details here:  
https://www.sanity.io/docs/open-source-plan

We do need to choose a license for our repo, and I’m terrible at knowing which one is best for any project.  Happy to get some input on the correct license to use …

**Reply by juliaschiller150** [reply]
Remix uses an MIT license, as does React Router and Remix Auth. I've heard it's usually best to follow one's community standards. My two cents!

**Reply by grantnorwood** [reply]
The MIT license was added in https://github.com/remix-austin/remixaustin-com/pull/74.

@brookslybrand, the [Sanity OSS plan](https://www.sanity.io/docs/open-source-plan) also requires us to have a Code of Conduct.  Do I remember you saying that you had been thinking about this?

I bet we could totally ~~ripoff~~ borrow the [Remix CoC](https://github.com/remix-run/remix/blob/main/CODE_OF_CONDUCT.md).

**Reply by brookslybrand** [reply]
Sweet! Thanks Grant. And yeah, I think that CoC should be great. Can't imagine why someone would mind us using their Code of Conduct since they whole point of them is to encourage good behavior in a shared project

**Reply by grantnorwood** [reply]
I created https://github.com/remix-austin/remixaustin-com/pull/75 to add the Code of Conduct.

**Comment by brookslybrand** [comment]
Calling bankruptcy on all discussions and issues today. Going to start revamping the direction of this project, so stay tuned 🚀
